pkg/cmd/playback.go

```diff
@@ -0,0 +1,245 @@
+
+	// --- Validate command-line args
+	// Check that mode is valid
+	if pc.mode != playback.Auto && pc.mode != playback.Record && pc.mode != playback.Replay {
+		return fmt.Errorf(
+			"\"%v\" is not a valid mode. It must be either \"%v\", \"%v\", or \"%v\"",
+			pc.mode, playback.Auto, playback.Record, playback.Replay)
+	}
+
+	// Check that cassette root directory is valid
+	var absoluteCassetteDir string
+	var err error
+	if filepath.IsAbs(pc.cassetteDir) {
+		absoluteCassetteDir = pc.cassetteDir
+	} else {
+		absoluteCassetteDir, err = filepath.Abs(pc.cassetteDir)
+		if err != nil {
+			return fmt.Errorf("Error with --cassette-root-dir: %w", err)
+		}
+	}
+
+	handle, err := os.Stat(absoluteCassetteDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return fmt.Errorf("the directory \"%v\" does not exist. Please create it, then re-run the command", absoluteCassetteDir)
+		}
+		return fmt.Errorf("Unexpected error when checking --cassette-root-dir: %w", err)
+	}
+
+	if !handle.Mode().IsDir() {
+		return fmt.Errorf("The provided `--cassette-root-dir` option is not a valid directory: %v", absoluteCassetteDir)
+	}
+
+	// TODO: disable HTTPS for now. it needs more work to be able to run in a released version of the stripe-cli
+	// eg: how should we package cert.pem and key.pem? (see stripe-mock's implementation)
+	pc.serveHTTPS = false
+
+	// --- Start up the playback HTTP server
+	// TODO: `playback` should handle setup (and teardown) of `stripe listen` as wcorell
+	addressString := pc.address
+	remoteURL := pc.apiBaseURL
+
+	var webhookAddress string
+	if pc.serveHTTPS {
+		webhookAddress = "https://" + pc.webhookAddress
+	} else {
+		webhookAddress = "http://" + pc.webhookAddress
+	}
+
+	httpWrapper, err := playback.NewRecordReplayServer(remoteURL, webhookAddress, absoluteCassetteDir)
+	if err != nil {
+		return nil
+	}
+
+	server := httpWrapper.InitializeServer(addressString)
+
+	if pc.serveHTTPS {
+		certFilepath := "pkg/playback/cert.pem"
+		keyFilepath := "pkg/playback/key.pem"
+		_, err := os.Stat(certFilepath)
+		if err != nil {
+			return fmt.Errorf("Error when loading cert.pem, are you sure it exists?: %w", err)
+		}
+
+		_, err = os.Stat(certFilepath)
+		if err != nil {
+			return fmt.Errorf("Error when loading key.pem, are you sure it exists?: %w", err)
+		}
+
+		go func() {
+			server.ListenAndServeTLS(certFilepath, keyFilepath)
+		}()
+	} else {
+		go func() {
+			server.ListenAndServe()
+		}()
+	}
+
+	var fullAddressString string
+	if pc.serveHTTPS {
+		fullAddressString = "https://" + addressString
+	} else {
+		fullAddressString = "http://" + addressString
+	}
+
+	tr := &http.Transport{
+		// TODO: (is this okay?) create a custom client to skip verifying our self-signed HTTPS certificate
+		// TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+
+	// --- Use the flag values to configure the server (over HTTP)
+	resp, err := client.Get(fullAddressString + "/pb/mode/" + pc.mode)
```

Comment:
Instead of hitting the configuration endpoints after the server starts up, I think it would be more natural to pass in this configuration initially when we call `NewRecordReplayServer`. That way we don't have to worry about `err != nil` or `defer resp.Body.Close`)